### PR TITLE
Post PnR static timing analysis 

### PR DESCRIPTION
**Pull request by jack-melchert**

### CODE REVIEW

**Review by Kuree — CHANGES_REQUESTED** [review]

File: archipelago/sta.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    print("\tCritical Path:", max_delay, "ns")
+    print(f"\t{max_node}", "glb:", timing_info[max_node].glbs,
+          "horiz hops:", timing_info[max_node].hhops,
+          "up hops:",  timing_info[max_node].uhops,
+          "down hops:",  timing_info[max_node].dhops,
+          "pes:", timing_info[max_node].pes,
+          "mems:", timing_info[max_node].mems, "\n")
+
+
+def load_id_to_name(packed_filename):
```
[the remark this answers is not in the document]
Comment (jack-melchert) [reply archipelago/sta.py:128]: pythunder.io.load_id_to_name doesn't seem to exist anymore, have you tried to use it recently?

File: archipelago/pnr_graph.py [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+                if self.get_node(node).type_ == "tile" and self.get_node(node).tile_id[0] == 'r':
+                    regs.append(node)
+            self.regs = regs
+        return self.regs
+
+    def get_shift_regs(self):
+        if not self.shift_regs:
+            regs = []
+            for node in self.nodes:
+                if self.get_node(node).type_ == "tile" and self.get_node(node).tile_id[0] == 'r' and "d_reg_" in self.id_to_name[node]:
```
[the remark this answers is not in the document]
Comment (jack-melchert) [reply archipelago/pnr_graph.py]: Yeah ill move to enum types if we keep using these graph tools. If we move to cyclone than this won't be used anyways.

File: archipelago/sta.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kuree) [reply archipelago/sta.py:122]: Oops. I might have removed the python binding. I will add it back as soon as possible after I'm done with the current stuff I have.

[the remark this answers is not in the document]
Comment (Kuree) [reply archipelago/sta.py:122]: Okay pushed the fix that adds back these IO functions. Can you try again?

File: archipelago/sta_delays.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (Kuree) [inline archipelago/sta_delays.json:1]: Can you make sure if this file is packaged properly when you do `python setup.py bdist_wheel`?

**Review by Kuree — CHANGES_REQUESTED** [review]

File: archipelago/sta.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jack-melchert) [reply archipelago/sta.py]: Looks fixed now

File: archipelago/sta_delays.json (hunk 3 above)
Comment (jack-melchert) [reply archipelago/sta_delays.json:1]: I added sta_delays to setup.py that is now packaged when running `python setup.py bdist_wheel`

Comment (jack-melchert) [reply archipelago/sta_delays.json:1]: I've never used bdist_wheel before though so let me know if I did it wrong

Comment (Kuree) [reply archipelago/sta_delays.json:1]: You forgot the `include_package_data = True`. I've pushed the changes.

**Review by Kuree — APPROVED** [review]

**Pull request merged**